tests/build/build.go

```diff
@@ -760,7 +763,7 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
 
 				It(fmt.Sprintf("the PipelineRun should eventually finish successfully for component %s", componentName), func() {
 					Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, "",
-						f.AsKubeAdmin.TektonController, &has.RetryOptions{Retries: 2, Always: true})).To(Succeed())
+						f.AsKubeAdmin.TektonController, &has.RetryOptions{Retries: 2, Always: true}, nil)).To(Succeed())
```

Comment:
Trying to understand how we decide when to pass "nil" instead of a PLR object.

Is it when we know that the LOC after this call uses an older PLR variable?

So, you passed nil here because we do call `GetComponentPipelineRun()` on line 783 below, so it will fetch and use the latest PLR. Correct?

Reply:
https://github.com/redhat-appstudio/e2e-tests/pull/1139/commits/2c5e0bba905fa792bd35d6dd4f32f36f981cdc31

Reply:
Thank you! like the fn description